packages/mongodb-mql-model/src/docs/md/bson-type/bson-type.md

```diff
@@ -0,0 +1,151 @@
+# MQL BSON Type
+-----------
+
+## Abstract
+
+This specification documents the different kinds of BSON types and how they are related to the
+original source code of an [MQL Query](../mql-query/mql-query.md). This document aims to provide
+information about the behaviour of dialects and linters on the computation of the original
+expression BSON type.
+
+## META
+
+The keywords "MUST", "MUST NOT", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY"
+and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+## Specification
+
+[BSON](https://bsonspec.org/spec.html) is a binary format that is used to communicate between the
+MongoDB Client (through a driver) and a MongoDB Cluster. MQL BSON (from now on we will just say BSON) 
+is a superset of the original BSON types.
+
+A BSON Type represents the data type inferred from the original source code or from a MongoDB sample
+of documents. A BSON Type MUST be consumable by a MongoDB Cluster and it's serialization MUST be
+BSON 1.1 compliant.
+
+### Primitive BSON Types
+
+#### [BsonString](/main/packages/mongodb-mql-model/src/main/kotlin/com/mongodb/jbplugin/mql/BsonType.kt#L58)
+
+A BsonString defines a sequence of characters independently of its locale or encoding. A BsonString MUST be
+encodable to a UTF-8 String.
```

Comment:
I think what these sentences boil down to "A BonString is a sequence of Unicode characters."